### ISSUE TITLE
Couldn't get the plugin to install

### ISSUE THREAD

**Issue by Rich28** [expectingtofly/LMS_BBC_Sounds_Plugin#162]
Previously had this great plugin working on a Raspberry Pi 2, which yep is a bit tad dated now, but all worked with LMS 8.3.1.  However, I've just had to do a clean complete re-install of the Pi, and while I've got the LMS server fully working again I can't for the life of me install the BBC Sounds app.  Maybe I'm going mad and I'm missing something obvious.  I've selected the app numerous times from the list of plugins and I know there's a requirement for me to sign into my BBC account, but I don't seem to be getting anywhere...

**Comment by expectingtofly** [comment]
I presume you are selecting it from the 3rd party plugins on the plugin tab in the lms settings, then clicking 'apply'?

Does it then say "restart server", but the plugin doesn't install after the LMS server is restarted?

Is there anything in the server.log?   Go to advanced->logging then look at the server.log

**Comment by Rich28** [comment]
Big thanks for getting back to me.  I did click on 'apply', I re-started the server  and even rebooted the pi, but no joy.  Thanks for suggesting to check the server log.  There are several entries in it that caught my eye, each one similar to this:- 

 `[23-12-22 13:00:29.8130] Slim::Utils::PluginDownloader::_installDownload (216) digest does not match /var/lib/squeezeboxserver/cache/DownloadedPlugins/BBCSounds.zip - BBCSounds will not be installed: expected d4affb806955ad948ad9530c6e09eacc87eb73d0, got 2ceb465e1e32d14e7cc6ac0cc86f4d1f51515015`

**Comment by expectingtofly** [comment]
Gosh that is a bit odd.  It looks like it is only partially downloaded the
plugin.    Is the second digest number always? '
2ceb465e1e32d14e7cc6ac0cc86f4d1f51515015' or does it change on each attempt
?

I wonder if you have enough room on your pi, are you installing
picoreplayer as the os for hosting LMS? If so, did you remember to expand
the filesystem?

**Comment by Rich28** [comment]
Hi, thanks for getting back to me again.  The sequence of numbers vaired, here for example is a bit more of the log file: -

`[23-12-22 12:59:14.5379] Slim::Utils::PluginDownloader::_installDownload (216) digest does not match /var/lib/squeezeboxserver/cache/DownloadedPlugins/BBCSounds.zip - BBCSounds will not be installed: expected d4affb806955ad948ad9530c6e09eacc87eb73d0, got 2ceb465e1e32d14e7cc6ac0cc86f4d1f51515015
[23-12-22 13:00:29.8130] Slim::Utils::PluginDownloader::_installDownload (216) digest does not match /var/lib/squeezeboxserver/cache/DownloadedPlugins/BBCSounds.zip - BBCSounds will not be installed: expected d4affb806955ad948ad9530c6e09eacc87eb73d0, got 2ceb465e1e32d14e7cc6ac0cc86f4d1f51515015
[23-12-22 13:00:58.4400] Slim::Networking::SqueezeNetwork::_error (421) Unable to login to SN: Invalid mysqueezebox.com username or password.
[23-12-22 13:01:13.4068] Slim::Utils::PluginDownloader::_installDownload (216) digest does not match /var/lib/squeezeboxserver/cache/DownloadedPlugins/UPnPBridge.zip - UPnPBridge will not be installed: expected 31177db9088a3036cbb1cf51562921fa49104043, got 6f8c61de98598dfa07af0d6b347de9aaf85ded56
[23-12-22 13:01:42.4312] Slim::Utils::PluginDownloader::_installDownload (216) digest does not match /var/lib/squeezeboxserver/cache/DownloadedPlugins/BBCSounds.zip - BBCSounds will not be installed: expected d4affb806955ad948ad9530c6e09eacc87eb73d0, got 2ceb465e1e32d14e7cc6ac0cc86f4d1f51515015
[23-12-22 13:01:52.5335] Slim::Utils::PluginDownloader::_installDownload (216) digest does not match `

I did check the filesystem, which seems properly inflated. Then I noticed a similar message was reported in the log file, relating to UPnPBridge was correctly installed.  Something strange might be going on here, well it is Xmas!! 

 Anyway, I was able to copy the folder appropriately called 'BBCSounds' (as situated in 'var/lib/squeezeboxsever/cache/InstalledPlugins/Plugins') across from my old install, which I guess is bit of hack, but the app appears to be working now.

**Comment by expectingtofly** [comment]
Ok, thanks.  All the hash checks are consistently wrong.  There is obviously something very wrong with your installation and how it is calculating the hash checks.  That will prevent you from installing or updating any plugins from the plugin repository. Which is obviously not really workable.  Something has gone very wrong with the install.  I would recommend using the 32 bit install not the 64 bit if you are installing picoreplayer?

Are you using picoreplayer?  If so it it might be worth posting on the picoreplayer thread, they will know what has gone wrong I expect.  https://forums.slimdevices.com/forum/user-forums/linux-unix?3-Linux-Unix=

**Comment by expectingtofly** [comment]
@Rich28 Coincidentally, I had a similar report on the support forum. It was caused by some security software in their router : https://forums.slimdevices.com/forum/user-forums/3rd-party-software/109758-announce-bbc-sounds-plugin?p=1663071#post1663071

**Comment by Rich28** [comment]
Hi, I really appreicate your input on this.  Whilst munching on a mince pie and indulging in the other festivities I've been  thinking about my issue, which cameabout when I did the re-install on the Pi.  I now realise that wasn't to a version of Bullseye, but was instead to Pi OS Bookworm, which has only just come out.  I've just reverted back to a Bullseye version of Pi OS and I'm really pleased to report that the app is now working as per normal again.  Sorry, I should have thought about the possibility Pi OS Bookworm being buggy before contacting you....
